src/System.Globalization.Calendars/tests/CalendarHelpers.cs

```diff
@@ -356,12 +357,12 @@ public static void ToDateTime_Invalid(Calendar calendar)
         public static void ToFourDigitYear_Invalid(Calendar calendar)
         {
             Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.ToFourDigitYear(-1));
-            // JapaneseCalandar allows any inputs below the max year
-            if (!(calendar is JapaneseCalendar))
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.ToFourDigitYear(MaxCalendarYearInEra(calendar, MaxEra(calendar)) + 1));
+            
+            if (!(calendar is JapaneseLunisolarCalendar))
```

Comment:
why we are specializing JapaneseLunisolarCalendar? I think we should have MinCalendarYearInEra be reliable even with JapaneseLunisolarCalendar